src/SNLS_device_forall.h

```diff
@@ -126,9 +251,10 @@ namespace snls {
    /// limitation of this wrapper is that the lambda captures can
    /// only capture functions / variables that are publically available
    /// if this is called within a class object.
-   template <const int NUMTHREADS, typename DBODY, typename HBODY>
-   inline void SNLS_ForallWrap(const int st,
+   template <const int NUMBLOCKS, const bool ASYNC = false, typename DBODY, typename HBODY>
+   inline rrese SNLS_ForallWrap(const int st,
```

Comment:
This function gives a warning about unused variable `d_body`. Might we consider changing this to prevent that warning?  One possibility is to have to forall wrappers, one for the GPUs and one for the CPUs.  That way all the function arguments are used.

Reply:
We can just wrap the d_body around a variation of this macro. I'd need to test this, but's a variation of something I've used in other codes for unused variables and last time I checked it worked across a wide range of compilers fairly well.

```c++
#if defined(UNUSED_CPU) && defined(__snls_gpu_active__)
# define UNUSED_CPU(x) x
#else
#if defined(UNUSED_CPU)
#elif defined(__GNUC__)
# define UNUSED_CPU(x) UNUSED_ ## x __attribute__((unused))
#else
# define UNUSED_CPU(x) x
#endif
#endif

```

Reply:
Added an unused variable similar to the above that should fix the warning. Although, I ended up calling things `UNUSED_GPU` which may or may not be the best name, and I'm open to updating the name.